Shopify/blockchain-components
# [Tailwind] Removes preflight plugin from Tailwind config

## ℹ️ What is the context for these changes?


Removes Tailwind preflight core plugin and addresses any inconsistencies that were a result of no longer having preflight.

> Why do we want to remove the preflight plugin?

Preflight [adds CSS](https://github.com/tailwindlabs/tailwindcss/blob/master/src/css/preflight.css) to the root of the stylesheet that has an impact on the consuming application. This is undesirable as it affects any hard work that developers have done up to the point of utilizing our components and also could drastically impact the display of a storefront that is integrating our components via a Shopify theme app extension.

## 🕹️ Demonstration



<details><summary>Create React App</summary>

| State | Image |
| - | - |
| Before (w/ preflight) | |
| Intermediary (w/o preflight) | |
| After (w/o preflight) | |

</details>

<details><summary>Next.js</summary>

| State | Image |
| - | - |
| Before (w/ preflight) | |
| Intermediary (w/o preflight) | |
| After (w/o preflight) | |

</details>

<details><summary>Vite</summary>

| State | Image (no interaction) | Modal open |
| - | - | - |
| Before (w/ preflight) | | |
| Intermediary (w/o preflight) | | |
| After (w/o preflight) | <img width="1409" alt="Screenshot 2023-03-15 at 5 02 23 PM" src="https://user-images.githubusercontent.com/4250423/225453802-18899a5a-6a22-49a5-be33-5e9aebaa36a4.png" /> | <img width="1409" alt="Screenshot 2023-03-15 at 5 02 33 PM" src="https://user-images.githubusercontent.com/4250423/225453829-1c3f218c-d5e9-4d57-98a8-91528857c208.png" /> |

</details>



## 🎩 How can this be tophatted?


### Testing `@shopify/connect-wallet`

I'd strongly recommend testing the package in the Vite example application. You can also test in Storybook, however, the issue I originally identified with the preflight styles conflicting and overwriting styles was within the Vite example.

1. Pull this branch
2. Ensure you've run `yarn install` at the repository root directory
3. Navigate to the Vite example application
4. Ensure you do not have node modules or a `yarn.lock`
5. Run `yarn install` in the Vite example
6. Run `yarn dev`

### Testing `@shopify/tokengate`

Since the tokengate package is not yet included in any examples, it is easiest to test it in Storybook.
